src/components/Header/Header.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todosToRender: Todo[] | undefined;
```

Comment:
Change the type of todosToRender from 'Todo[] | undefined' to 'Todo[]'. This will make the code less prone to errors.